src/graph/network.cc

```diff
@@ -88,6 +113,54 @@ void MsgMeta::Deserialize(char* buffer, int64_t size) {
   CHECK_EQ(data_size, size);
 }
 
+char* KVStoreMsg::Serialize(int64_t* size) {
+  char* buffer = nullptr;
+  int64_t buffer_size = 0;
+  buffer_size += sizeof(this->msg_type);
+  buffer_size += sizeof(this->rank);
+  if (!this->name.empty()) {
+    buffer_size += sizeof(this->name.size());
+    buffer_size += this->name.size();
+  }
+  buffer = new char[buffer_size];
```

Comment:
Can you put a comment here? In the future, we should have a better memory management. allocating a large chunk of memory can be very expensive.